src/main/java/com/google/sps/data/OrderHandler.java

```diff
@@ -112,14 +112,17 @@ public void addOrderToDatastore(LibraryPoint library, List<String> bookIds, Stri
     for (LibraryPoint library : libraryBookIds.keySet()) {
       ArrayList<String> booksIds = libraryBookIds.get(library);
       addOrderToDatastore(library, booksIds, userId, address);
-      booksManager.removeBooksFromLibrary(library, booksIds);
+      boolean flag = booksManager.removeBooksFromLibrary(library, booksIds);
+      if (flag == false) {
+        throw new BadRequestException("No stock available!");
```

Comment:
Bad request is generally used when the client request is "syntactically incorrect".
https://docs.oracle.com/javaee/6/api/javax/servlet/http/HttpServletResponse.html

That means the request is invalid regardless of server state.

Maybe DataNotFoundException would be better here?

Reply:
Done, it throws DataNotFoundException now